### PR TITLE
Add support for sending requests with channel.

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/173/
Test PASSed.

**Comment by mhilton** [comment]
:+1:

File: csclient/csclient.go [hunk 1]
```diff
@@ -535,6 +544,11 @@ func (c *Client) DoWithBody(req *http.Request, path string, body io.ReadSeeker)
 	if err != nil {
 		return nil, errgo.Mask(err)
 	}
+	if c.channel != params.NoChannel {
+		values := u.Query()
+		values.Add("channel", string(c.channel))
```
Comment (rogpeppe) [inline csclient/csclient.go]: Hmm, wouldn't this be better as values.Set?

Comment (frankban) [inline csclient/csclient.go]: Done.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/174/
Test PASSed.

**Comment by frankban** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmrepo-merge

**Pull request merged**